vsphere/datadog_checks/vsphere/api.py

```diff
@@ -115,14 +115,16 @@ def smart_connect(self):
             # Remove type ignore when this is merged https://github.com/python/typeshed/pull/3855
             context = ssl.SSLContext(ssl.PROTOCOL_TLS)  # type: ignore
             context.verify_mode = ssl.CERT_NONE
-        elif self.config.ssl_cafile or self.config.ssl_capath:
+        elif self.config.ssl_capath or self.config.ssl_cafile:
             # Remove type ignore when this is merged https://github.com/python/typeshed/pull/3855
             context = ssl.SSLContext(ssl.PROTOCOL_TLS)  # type: ignore
             context.verify_mode = ssl.CERT_REQUIRED
             # `check_hostname` must be enabled as well to verify the authenticity of a cert.
             context.check_hostname = True
-            context.load_verify_locations(cafile=self.config.ssl_cafile, capath=self.config.ssl_capath)
-
+            if self.config.ssl_capath:
+                context.load_verify_locations(cafile=None, capath=self.config.ssl_capath)
+            else:
+                context.load_verify_locations(cafile=self.config.ssl_cafile, capath=None)
```

Comment:
To align with
https://github.com/DataDog/integrations-core/blob/7e5f6a589d98edf538b056ad6b59231fb5751c68/datadog_checks_base/datadog_checks/base/utils/tls.py#L123-L128